src/test/transaction_tests.cpp

```diff
@@ -897,37 +912,55 @@ BOOST_AUTO_TEST_CASE(TxV5)
         BOOST_CHECK_EQUAL(tx.GetHash().GetHex(), test[1].getValStr());
         BOOST_CHECK_EQUAL(tx.GetAuthDigest().GetHex(), test[2].getValStr());
 
+        UniValue amountsArr = test[3].get_array();
+        UniValue scriptCodesArr = test[4].get_array();
+        std::vector<CAmount> amounts;
+        std::vector<CScript> scriptCodes;
+        std::vector<CTxOut> allPrevOutputs;
+        if (tx.IsCoinBase()) {
+            BOOST_CHECK(amountsArr.empty());
+            BOOST_CHECK(scriptCodesArr.empty());
+        } else {
+            BOOST_CHECK_EQUAL(amountsArr.size(), tx.vin.size());
+            BOOST_CHECK_EQUAL(scriptCodesArr.size(), tx.vin.size());
+
+            for (size_t inpIdx = 0; inpIdx < tx.vin.size(); inpIdx++) {
+                amounts.push_back(amountsArr[inpIdx].get_int64());
+                auto scriptCodeBytes = ParseHex(scriptCodesArr[inpIdx].get_str());
+                scriptCodes.push_back(CScript(scriptCodeBytes.begin(), scriptCodeBytes.end()));
+                allPrevOutputs.emplace_back(amounts[inpIdx], scriptCodes[inpIdx]);
+            }
+        }
+
         // ZIP 244: Check the signature digests.
         unsigned int nIn = NOT_AN_INPUT;
-        if (!test[3].isNull()) {
-            nIn = test[3].get_int();
+        if (!test[5].isNull()) {
+            nIn = test[5].get_int();
         }
 
         CScript scriptCode;
-        if (!test[4].isNull()) {
-            auto scriptCodeBytes = ParseHex(test[4].get_str());
-            scriptCode = CScript(scriptCodeBytes.begin(), scriptCodeBytes.end());
-        }
-
         CAmount amount;
-        if (!test[5].isNull()) {
-            amount = test[5].get_int64();
+        if (nIn != NOT_AN_INPUT) {
+            scriptCode = scriptCodes[nIn];
+            amount = amounts[nIn];
         }
 
+        const PrecomputedTransactionData txdata(tx, allPrevOutputs);
+
         BOOST_CHECK_EQUAL(
             SignatureHash(
-                scriptCode, tx, nIn,
+                scriptCode, tx, NOT_AN_INPUT,
                 SIGHASH_ALL,
-                amount, *tx.GetConsensusBranchId()
+                amount, *tx.GetConsensusBranchId(), txdata
             ).GetHex(),
             test[6].getValStr());
 
         if (!test[7].isNull()) {
             BOOST_CHECK_EQUAL(
                 SignatureHash(
                     scriptCode, tx, nIn,
-                    SIGHASH_NONE,
-                    amount, *tx.GetConsensusBranchId()
+                    SIGHASH_ALL,
+                    amount, *tx.GetConsensusBranchId(), txdata
```

Comment:
I was pretty puzzled by this until I realized that this was just a consequence of where the new tests were added in the test vector.